method.rb

```diff
@@ -0,0 +1,181 @@
+module Enumerable # rubocop:disable Style/ModuleLength
+  def my_each
+    i = 0
+    arr = to_a
+    return to_enum(:my_each) unless block_given?
+
+    while i < arr.size
+      yield arr[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    arr = to_a
+    return to_enum(:my_each_with_index) unless block_given?
+
+    while i < arr.size
+      yield arr[i], i
+      i += 1
+    end
+
+    self
+  end
+
+  def my_select
+    new_arr = []
+    return to_enum(:my_select) unless block_given?
+
+    my_each do |num|
+      new_arr.push(num) if yield num
+    end
+
+    new_arr
+  end
+
+  def my_all?(arg = nil) # rubocop:disable Style/MethodLength
+    stat = true
+
+    if arg
+      if arg.class == Class
+        my_each do |num|
+          stat = num.is_a?(arg)
```

Comment:
- [ ] The stat should be true only if all of the items in the enumerable are of the given Class.
Make sure it returns false the first time if finds something that is not from the given Class.